File: backend/src/RssFeedParser/Models/Feed.cs
namespace RssFeedParser.Models;

public record Feed
{
    public string Title { get; set; } = string.Empty;
    public string Link { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public List<FeedItem> Items { get; set; } = new();
}

public record FeedItem
{
    public string Title { get; set; } = string.Empty;
    public string Link { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string PubDate { get; set; } = string.Empty;
}

File: backend/internal/repository/models/feed.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
)

type Feed struct {
	ID          bson.ObjectID `bson:"_id,omitempty"`
	Title       string        `bson:"title"`
	Description string        `bson:"description"`
	Link        string        `bson:"link"`
	Articles    []Articles    `bson:"articles"`
	CreatedAt   time.Time     `bson:"created_at" json:"created_at"`
}

type Articles struct {
	URL   string `bson:"url"`
	Title string `bson:"title,omitempty"`
}
